tests/test_pf7.py

```diff
@@ -0,0 +1,433 @@
+        with patch(
+            "malariagen_data.pf7.init_filesystem",
+            return_value=["fs", self.test_data_path],
+        ):
+            pf7_mock_fs = Pf7(self.test_data_path, data_config=self.test_config_path)
+        pf7_mock_fs.open_zarr()
+        mock_safestore.assert_called_once_with(fs="fs", path=self.test_zarr_path)
+        mock_zarr.assert_called_once_with(store="Safe store object")
+
+    @patch("malariagen_data.pf7.init_zarr_store", return_value="Safe store object")
+    @patch("malariagen_data.pf7.zarr.open_consolidated")
+    def test_open_zarr_uses_cache(self, mock_zarr, mock_safestore):
+        with patch(
+            "malariagen_data.pf7.init_filesystem",
+            return_value=["fs", self.test_data_path],
+        ):
+            pf7_mock_fs = Pf7(self.test_data_path, data_config=self.test_config_path)
+        pf7_mock_fs.open_zarr()
+        pf7_mock_fs.open_zarr()
+        mock_safestore.assert_called_once_with(fs="fs", path=self.test_zarr_path)
+        mock_zarr.assert_called_once_with(store="Safe store object")
+
+    def test_subset_extended_dictionary(self):
+        (
+            actual_variants_dictionary,
+            actual_calldata_dictionary,
+        ) = self.test_pf7_class._subset_extended_dictionary(
+            ["DP", "GQ", "AC", "AF", "ANN_Feature_Type"]
+        )
+        expected_calldata_dictionary = {
+            "DP": [DIM_VARIANT, DIM_SAMPLE],
+            "GQ": [DIM_VARIANT, DIM_SAMPLE],
+        }
+        expected_variants_dictionary = {
+            "DP": [DIM_VARIANT],
+            "AC": [DIM_VARIANT, DIM_ALT_ALLELE],
+            "AF": [DIM_VARIANT, DIM_ALT_ALLELE],
+            "ANN_Feature_Type": [DIM_VARIANT, DIM_ALT_ALLELE],
+        }
+        self.assertEqual(actual_variants_dictionary, expected_variants_dictionary)
+        self.assertEqual(actual_calldata_dictionary, expected_calldata_dictionary)
+
+    def test_subset_extended_dictionary_raises_error(self):
+        with self.assertRaises(ValueError):
+            self.test_pf7_class._subset_extended_dictionary(
+                ["DP", "AC", "BAD_VARIABLE"]
+            )
+
+    def test_add_coordinates(self):
+        var_names_for_outputs = {
+            "POS": "position",
+            "CHROM": "chrom",
+            "FILTER_PASS": "filter_pass",
+        }
+        actual_coordinates = self.test_pf7_class._add_coordinates(
+            self.test_zarr_root, True, "native", var_names_for_outputs
+        )
+        self.assertEqual(
+            list(actual_coordinates.keys()),
+            ["variant_position", "variant_chrom", "sample_id"],
+        )
+
+        values = [actual_coordinates[k] for k in actual_coordinates]
+        for value in values:
+            assert isinstance(value[1], da.Array)
+        dimensions = [i[0] for i in values]
+        self.assertEqual(
+            dimensions,
+            [["variants"], ["variants"], ["samples"]],
+        )
```

Comment:
This tests behaviour of `_add_coordinates()` private method which is fair enough, although personally I would stick to testing behaviour of public methods only, because these will give you the same coverage of behaviour, and you give yourself freedom to refactor internally later without having to rewrite tests.